docs/uptime/uptime.md

```diff
@@ -0,0 +1,289 @@
+| 1 year     | 1 minute        | 1                | 524,160     |
+
+| Time range | Check frequency | Number of probes | Executions  |
+|------------|-----------------|------------------|-------------|
+| 1 hour     | 1 minute        | 3                | 180         |
+| 1 day      | 1 minute        | 3                | 4,320       |
+| 1 week     | 1 minute        | 3                | 30,240      |
+| 1 year     | 1 minute        | 3                | 1,572,480   |
+
+| Time range | Check frequency | Number of probes | Executions  |
+|------------|-----------------|------------------|-------------|
+| 1 hour     | 15 seconds      | 20               | 4800        |
+| 1 day      | 15 seconds      | 20               | 115,200     |
+| 1 week     | 15 seconds      | 20               | 806,400     |
+| 1 year     | 15 seconds      | 20               | 41,932,800  |
+
+3. A check might have multiple configurations during the time period we are querying. This could be a change in frequency, the number of probes or the check's definition of uptime itself. This could lead to a change in the number of time points and the number of them which are successful.
+
+## How we calculate uptime with PromQL
+
+Taking into account all of the above, our uptime calculation is the following:
+
+```
+clamp_max(sum(max_over_time(probe_success{job="$job", instance="$instance"}[$frequencyInSeconds])), 1)
+```
+
+With the additional options:
+
+```
+maxDataPoints: 8000
+minInterval: $frequencyInSeconds
+```
+
+To explain the individual parts of the query, starting from the innermost part:
+
+### probe_success{job="$job", instance="$instance"}
+
+`probe_success` is a [Prometheus gauge metric](https://prometheus.io/docs/concepts/metric_types/#gauge) reported by each of our probes. It can only have a value of 0 or 1. 0 represents a failure and 1 represents a success. We add `job` and `instance` label filters as these two combined are how we uniquely identify a check. `probe` and `config_version` labels are also available on this metric but are not explicitly used in the uptime calculation. It is worth noting they exist on the metric because if we modify the base query we can end up with very different results depending on the scenario _[See the example below](#what-happens-if-we-remove-the-max_over_time-aggregation-from-the-query)_.
+
+### max_over_time({{_metric_}}[$frequencyInSeconds])
+
+`max_over_time` is a [Prometheus aggregation function](https://prometheus.io/docs/prometheus/latest/querying/functions/#aggregation_over_time) that returns the maximum value of a metric over a given time range, in this case a check's frequency in seconds: `$frequencyInSeconds`.
+
+We use (`$frequencyInSeconds`) to determine what the 'look back' time should be (which gives us a 'time point'). For example, if the check is running every 15 seconds, the step would be 15 seconds. If the check is running every minute, the step would be 60 seconds.
```

Comment:
I think Marcelo also covers this in his comments, but by using frequencyInSeconds do we create the possibility of getting no samples in a range?  

I think the assumption of a data point in the range will hold much of the time, but what happens when it does not?  I'm thinking of script-based tests in particular, there is variability because of script run duration, lag, and retries that can happen on certain types of failure.